src/main/java/com/cloudbees/plugins/credentials/CredentialsParameterDefinition.java

```diff
@@ -125,29 +126,27 @@ public String getDisplayName() {
         public ListBoxModel doFillCredentialTypeItems() {
             ListBoxModel result = new ListBoxModel();
             result.add("Any", StandardCredentials.class.getName());
-            for (Descriptor<Credentials> d : CredentialsProvider.allCredentialsDescriptors()) {
-                if (!(d instanceof CredentialsDescriptor)) {
+            for (CredentialsDescriptor d : CredentialsProvider.allCredentialsDescriptors()) {
```

Comment:
No, the original code was correct, it is actually `allCredentialsDescriptors` which is unsafe.

Reply:
Reverted (incl. the two comments below).